src/stream/src/common/table/state_table.rs

```diff
@@ -650,6 +718,8 @@ where
     /// Insert a row into state table. Must provide a full row corresponding to the column desc of
     /// the table.
     pub fn insert(&mut self, value: impl Row) {
+        self.watermark_min_value_cache
```

Comment:
Should we do static dispatch, if state table has watermark for this?

In the case where state_table has watermark, it has to update this for every row. Which also seems very costly. (See the `handle_insert` function above to see costs).